src/App.tsx

```diff
@@ -15,48 +18,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_NAME: SortType = SortType.Name;
+const SORT_FIELD_LENGTH: SortType = SortType.Length;
+let isResetVisible = false;
+
+type Goods = string[];
+
+function getPreparedGoods(
+  goods: Goods,
+  { sortField, isReversed }: SortingParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    isResetVisible = true;
+
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    isResetVisible = true;
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setReversed] = useState<boolean>(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
+  function setReverseButtonAction(reverse: boolean) {
+    if (reverse) {
+      isResetVisible = false;
+
+      return setReversed(false);
+    }
+
+    return setReversed(true);
+  }
+
+  function resetSorting() {
+    setSortField(SortType.None);
+    setReversed(false);
+    isResetVisible = false;
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+          className={`button is-info ${classNames({ 'is-light': sortField !== SORT_FIELD_NAME })}`}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          className={`button is-success ${classNames({ 'is-light': sortField !== SORT_FIELD_LENGTH })}`}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          onClick={() => setReverseButtonAction(isReversed)}
+          className={`button is-warning ${classNames({ 'is-light': !isReversed })}`}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {isResetVisible
+        && (
+          <button
+            type="button"
+            onClick={resetSorting}
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
```

Comment:
```suggestion
        {isResetVisible && (
          <button
            type="button"
            onClick={resetSorting}
            className="button is-danger is-light"
          >
            Reset
          </button>
        )}
```

check formatting please